### PR TITLE
refactor(ivy): Update query-related comments

### PR DESCRIPTION
**Pull request by benlesh**
Just updating comments in query-related things to make it easier for the next person that has to grok this for the first time.

Also adds a demo from @mhevery to one of the query specs

Related #29031

### CODE REVIEW

**Review by AndrewKushnir — APPROVED** [review]
LGTM!

File: packages/core/src/render3/query.ts [hunk 1]
```diff
@@ -365,7 +365,8 @@ export function query<T>(
 /**
  * Refreshes a query by combining matches from all active views and removing matches from deleted
  * views.
- * Returns true if a query got dirty during change detection, false otherwise.
+ *
+ * @returns `true` if a query got dirty during change detection, `false` otherwise.
```
Comment (kara) [inline packages/core/src/render3/query.ts]: Can you update this based on my comment on your other pr? It's not quite accurate for static queries

**Review by kara — CHANGES_REQUESTED** [review]

**Review by mhevery — APPROVED** [review]

**Comment by mhevery** [comment]
Please address @kara's comment, LGTM otherwise.

File: packages/core/src/render3/query.ts (hunk 1 above)
Comment (benlesh) [reply packages/core/src/render3/query.ts]: I suspect you mean this comment? https://github.com/angular/angular/pull/29031#discussion_r265406725

**Review by kara — APPROVED** [review]
LGTM

**Comment by benlesh** [comment]
Caretaker: @kara and @mhevery are global approvers.

**Comment by matsko** [comment]
This made master fail. Please rebase.

**Pull request closed without merging**